auth/api/iam/api.go

```diff
@@ -236,12 +237,14 @@ func (r Wrapper) RetrieveAccessToken(_ context.Context, request RetrieveAccessTo
 		// return pending status
 		return RetrieveAccessToken200JSONResponse(token), nil
 	}
-	// delete access token from store
+	// access token is active, return to caller and delete access token from store
 	// change this when tokens can be cached
-	err = r.accessTokenClientStore().Delete(request.SessionID)
-	if err != nil {
-		return nil, err
-	}
+	defer func() {
+		err = r.accessTokenClientStore().Delete(request.SessionID)
```

Comment:
why do this in a deferred func? As you just log the error and return token in any case, there's no difference in doing it  deferred or not?

Reply:
This is a fallback for when it fail before it gets to the GetAndDelete method, to make sure the session is always burned

Reply:
defer calls the function when the current scope finishes, meaning after return at line 249, so there's nothing that can fail after this defer?

Reply:
my bad, I thought I had only added 1 defer function. Removed defer, but kept logging instead of failing when delete token fails. Since this is retrieval by the local node this is preferred over failing?